sympy/core/basic.py

```diff
@@ -367,49 +367,106 @@ def __ne__(self, other):
 
     def dummy_eq(self, other, symbol=None):
         """
-        Compare two expressions and handle dummy symbols.
+        Return True if ``self`` and ``other`` are the same (or
+        can be made so by mapping Dummy symbols in ``self`` to
+        Dummy symbols (default) in ``other``. Symbols that are
+        not actually Dummy symbols in ``other`` can be treated
+        as such by passing them -- one or more -- in ``symbol``.
+
+        If there are no Dummy symbols in ``self`` the result
+        can only be true if the expressions are structurally the
+        same in their use of free symbols.
```

Comment:
To me this seems like a separate function from `dummy_eq` that should have a different name.